### PR TITLE
update marlinfw.org ruby and jekyll

### PR DESCRIPTION
**Pull request by ellensp**
Update marlinfw.org to comply with Github pages dependency versions https://pages.github.com/versions/ 

I've tested this on local linux machine, it looks complete, but needs checking by someone familiar with every facet.. I haven't tested .github/workflows/jekyll-pub.yml

There are new docs coming from thisiskeithb on how to install locally for all the OS's

My notes: Install Ruby 3.3.4 on Ubuntu 
 
I Installed https://github.com/rbenv/rbenv as the os version is not recent, mostly following https://shape.host/resources/step-by-step-guide-installing-ruby-on-rails-with-rbenv-on-ubuntu-22-04

Enter these commands

```
  sudo apt update
  sudo apt install git curl libssl-dev libreadline-dev zlib1g-dev autoconf bison build-essential libyaml-dev libreadline-dev libncurses5-dev libffi-dev libgdbm-dev
  curl -fsSL https://github.com/rbenv/rbenv-installer/raw/HEAD/bin/rbenv-installer | bash

  echo 'export PATH="$HOME/.rbenv/bin:$PATH"' >> ~/.bashrc
  echo 'eval "$(rbenv init -)"' >> ~/.bashrc

  source ~/.bashrc

  rbenv install 3.3.4
  rbenv global 3.3.4
  ruby -v

  echo '# Install Ruby Gems to ~/gems' >> ~/.bashrc
  echo 'export GEM_HOME="$HOME/gems"' >> ~/.bashrc
  echo 'export PATH="$HOME/gems/bin:$PATH"' >> ~/.bashrc
  source ~/.bashrc

  gem install bundler
```
Get MarlinDocumentation with updated files from this PR

change into MarlinDocumentation directory and enter these commands

```
  bundle config set path 'vendor/bundle'
  bundle install
  bundle exec jekyll -v
  bundle exec jekyll serve --watch --incremental

```

### CODE REVIEW

**Comment by thisiskeithb** [comment]
> There are new docs coming from thisiskeithb oh how to install locally.

Yep. Over in https://github.com/MarlinFirmware/MarlinDocumentation/pull/575.

I have an Ubuntu 24.04 VM I planned to test these instructions in. I also need to confirm if the updated instructions in my PR (with modified version numbers) work the same in Windows and macOS still, so thanks a lot!

I kid, I kid. This is great news 😄 

Now we can focus on bringing everything else up to date...

**Comment by thisiskeithb** [comment]
One minor suggestion: Add a `.ruby-version` file in the project root with `3.3.4` in it so the Ruby version is switched automatically for those that have configured their system to do so.

**Comment by ellensp** [comment]
I was thinking more add it to .gitignore

File: instructions [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+echo 'export PATH="$HOME/gems/bin:$PATH"' >> ~/.bashrc
+source ~/.bashrc
+
+gem install bundler
+
+Get MarlinDocumentation
+
+git clone https://github.com/MarlinFirmware/MarlinDocumentation.git
+
+Update Gemfile with new gems versions
```
Comment (thinkyhead) [inline instructions:51]: All the DIFF stuff can be removed from this file.

Simply deleting `Gemfile.lock` and running `bundle install` in the folder should do all the right things for an older `vendor/bundle` folder. The `Gemfile` will already be in the correct state from doing the `git pull`. If there are any issues with `bundle install` not working, deleting the `vendor/bundle` folder and then running `bundle install` again should do the trick.

The README instructions for all platforms should now have everything covered all the way from the install of Ruby to preview of documentation. Every "install" of the documentation can be treated as a "fresh install" and if savvy contributors are maintaining work branches, they already know how to `git rebase` them.

**Pull request merged**